lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -2193,6 +2193,12 @@ public function setAssociationOverride($fieldName, array $overrideMapping)
 
         $mapping = $this->associationMappings[$fieldName];
 
+        if (isset($mapping['inherited']) && (count($overrideMapping) !== 1 || ! isset($overrideMapping['fetch']))) {
+            // TODO: Deprecate overriding the fetch mode via association override for 3.0,
+            // users should do this with a listener and a custom attribute/annotation
+            throw MappingException::illegalOverrideOfInheritedProperty($this->name, $fieldName);
+        }
```

Comment:
Should the parts that are about throwing be contributed to 2.8 instead? If we think there are people currently using this, we should probably not have it blow up in their face in a patch

Reply:
You are right, i should remove them for 2.7, and re-introduce them for 2.8